### PR TITLE
updated aerosol convective wet removal for GEFS-Aer 12.3

### PR DESCRIPTION
**Pull request by lipan-NOAA**
## Description

This change is only for production/GEFS_v12 branch.

### Issue(s) addressed

https://github.com/NOAA-EMC/fv3atm/issues/539

### CODE REVIEW

**Comment by lipan-NOAA** [comment]
@XianwuXue-NOAA @WalterKolczynski-NOAA who is the owner of production/GEFS_v12 ?

File: gfsphysics/GFS_layer/GFS_physics_driver.F90 [hunk 1]
```diff
@@ -3207,7 +3209,7 @@ subroutine GFS_physics_driver                                 &
       endif   ! end if_ldiag3d/lgocart/cplchm
 
       if (Model%lgocart .or. Model%cplchm) then
-        Coupling%dqdti(1:im,:) = 0._kind_phys
+        Coupling%dqdti(1:im,:) = zero
```
Comment (XianwuXue-NOAA) [inline gfsphysics/GFS_layer/GFS_physics_driver.F90]: Why you changed "0._kind_phys" to "zero"? If they are the same, I think it is better to keep the original "0._kind_phys"

File: gfsphysics/GFS_layer/GFS_physics_driver.F90 [hunk 2]
```diff
@@ -3590,7 +3592,7 @@ subroutine GFS_physics_driver                                 &
                              Model%clam_deep,   Model%c0s_deep,                    &
                              Model%c1_deep,  Model%betal_deep, Model%betas_deep,   &
                              Model%evfact_deep, Model%evfactl_deep,                &
-                             Model%pgcon_deep,  Model%asolfac_deep)
+                             Model%pgcon_deep,  Model%asolfac_deep, wet_deep) !lzhang
 !           if (lprnt) print *,' rain1=',rain1(ipr)
```
Comment (XianwuXue-NOAA) [inline gfsphysics/GFS_layer/GFS_physics_driver.F90:3596]: there are a lot of "!lzhang" in the new changes, I think if there is not necessary, then you can delete them or add more detailed description.

File: gfsphysics/physics/samfdeepcnv.f [hunk 3]
```diff
@@ -2704,7 +2707,13 @@ subroutine samfdeepcnv(im,ix,km,delt,itc,ntc,ntk,ntr,delp,
           do k = 1, km
             do i = 1, im
               if(cnvflg(i) .and. rn(i) > 0.) then
-                if (k <= kmax(i)) qtr(i,k,kk) = qaero(i,k,n)
+                !if (k <= kmax(i)) qtr(i,k,kk) = qaero(i,k,n)
```
Comment (XianwuXue-NOAA) [inline gfsphysics/physics/samfdeepcnv.f]: If this statement will not be used, I think it is better to clean them up.

File: gfsphysics/physics/samfshalcnv.f [hunk 4]
```diff
@@ -1756,7 +1758,13 @@ subroutine samfshalcnv(im,ix,km,delt,itc,ntc,ntk,ntr,delp,
           do k = 1, km
             do i = 1, im
               if(cnvflg(i) .and. rn(i) > 0.) then
-                if (k <= kmax(i)) qtr(i,k,kk) = qaero(i,k,n)
+                !if (k <= kmax(i)) qtr(i,k,kk) = qaero(i,k,n)
```
Comment (XianwuXue-NOAA) [inline gfsphysics/physics/samfshalcnv.f]: Same for this line.

**Comment by XianwuXue-NOAA** [comment]
@WalterKolczynski-NOAA Could you renew this PR and also add @zhanglikate? Thanks.

File: gfsphysics/physics/samfshalcnv.f (hunk 4 above)
Comment (zhanglikate) [reply gfsphysics/physics/samfshalcnv.f]: Agree.

File: gfsphysics/physics/samfdeepcnv.f (hunk 3 above)
Comment (zhanglikate) [reply gfsphysics/physics/samfdeepcnv.f]: Agree.

File: gfsphysics/GFS_layer/GFS_physics_driver.F90 (hunk 2 above)
Comment (zhanglikate) [reply gfsphysics/GFS_layer/GFS_physics_driver.F90:3596]: I suggest to remove "lzhang".

File: gfsphysics/GFS_layer/GFS_physics_driver.F90 (hunk 1 above)
Comment (zhanglikate) [reply gfsphysics/GFS_layer/GFS_physics_driver.F90]: I think this is version different, in my version , it used zero as default, nothing change. Better to keep your original version.

**Review by zhanglikate — APPROVED** [review]
Please update them as the comment.

File: gfsphysics/GFS_layer/GFS_physics_driver.F90 (hunk 1 above)
Comment (lipan-NOAA) [reply gfsphysics/GFS_layer/GFS_physics_driver.F90]: switch back to the original version

File: gfsphysics/GFS_layer/GFS_physics_driver.F90 (hunk 2 above)
Comment (lipan-NOAA) [reply gfsphysics/GFS_layer/GFS_physics_driver.F90:3596]: did it

File: gfsphysics/physics/samfdeepcnv.f (hunk 3 above)
Comment (lipan-NOAA) [reply gfsphysics/physics/samfdeepcnv.f]: did it

File: gfsphysics/physics/samfshalcnv.f (hunk 4 above)
Comment (lipan-NOAA) [reply gfsphysics/physics/samfshalcnv.f]: did it

**Comment by lipan-NOAA** [comment]
@XianwuXue-NOAA @WalterKolczynski-NOAA 
a new commit was pushed

**Review by WalterKolczynski-NOAA — APPROVED** [review]
It looks fine to me, but I have no expertise in the UFS physics code.

**Review by XianwuXue-NOAA — APPROVED** [review]
It looks fine to me. However, I am not familiar with the codes, @zhanglikate please double check the codes.

**Review by XianwuXue-NOAA — APPROVED** [review]

**Comment by junwang-noaa** [comment]
@WalterKolczynski-NOAA Do you want to merge this PR to GEFS_v12 branch?

**Comment by lipan-NOAA** [comment]
@junwang-noaa In my understanding, yes it is only merged to GEFS_v12 branch. @WalterKolczynski-NOAA please comment it.
Thanks

**Comment by WalterKolczynski-NOAA** [comment]
Yes, just GEFS

**Pull request merged**